fiftyone/core/annotation/validate_label_schemas.py

```
@@ -53,19 +62,24 @@ def validate_label_schemas(
         sample_collection, flatten=True
     )
     exceptions = []
-    for field in fields:
+    for field_name in fields:
         try:
-            if field not in all_fields:
-                raise ValueError(f"field '{field}' does not exist")
+            if allow_new_fields is False and field_name not in all_fields:
+                raise ValueError(f"field '{field_name}' does not exist")
```

Ruff (0.14.14)

Abstract `raise` to an inner function

(TRY301)